### DISCUSSION TITLE
flux bootstrap gitlab documentation is missing a few details

### DISCUSSION THREAD

**Discussion by crinjes** [fluxcd/flux2#2247] [General]
When trying to bootstrap v0.24.1 on a self-hosted GitLab instance (14.6.0-ee), I've run into multiple quirks that were not documented on the installation page, and all resulted in an obscure `authorization failed` error when cloning or pushing.

- The instance was only allowing SSH access to git. I had to change it to HTTPS+SSH. I only managed to discover this by searching the GitLab logs and trying the path myself. GitLab even returns a helpful `Git access over HTTP is not allowed` message in the body.
- Using a project access token, a user with email `project1234_bot@example.com` is created. I had to set `author-email` to this address. The default empty address is not allowed either.
- Thanks to the quirk above, I had to change the push rules that restrict the commit author's email accordingly.

**Comment by kingdonb** [comment]
What was the exact `flux bootstrap` command that you ran?

We have some gitlab-ce instances that we can try this out on, if these issues are specific to ee then it's important to get them documented just as well, though we might not be able to reproduce them.

@darkowlzz FYI
